py21cmsense/observatory.py

```diff
@@ -130,6 +133,20 @@ def from_yaml(cls, yaml_file):
                 "file, or convertible to a ndarray"
             )
 
+        # Mask out some antennas if a max_antpos is set in the YAML
+        max_antpos = data.pop("max_antpos", np.inf)
+        _n = len(antpos)
+        antpos = antpos[np.sum(np.square(antpos), axis=1) < max_antpos ** 2]
+
+        if max_antpos < np.inf:
+            logger.info(
+                f"Removed {_n - len(antpos)} antennas using given max_antpos={max_antpos} m."
+            )
+
+        # If we get only East and North coords, add zeros for the UP direction.
+        if antpos.shape[1] == 2:
+            antpos = np.hstack((antpos, np.zeros((len(antpos), 1))))
```

Comment:
While you're adding a maximum antenna position, is there a desire for a minimum one too? or a minimum uvw range maybe?

Reply:
There's already a `bl_min` and `bl_max` for the `Observation` (in meters units). It makes sense to have a minimum antenna position. I'll add that.

Reply:
OK, did that. But also realized it was only using those parameters from a YAML, not when instantiating the object directly. So I reshuffled it to use those parameters at the top-level.

Then I did `pre-commit autoupdate` and it changed a whole bunch of files. Sorry.